pkg/pillar/cmd/zedkube/zedkube.go

```diff
@@ -360,6 +529,11 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject, ar
 	zedkubeCtx.subEdgeNodeClusterConfig = subEdgeNodeClusterConfig
 	subEdgeNodeClusterConfig.Activate()
 
+	if len(subEdgeNodeClusterConfig.GetAll()) != 0 {
+		// Handle persistent existing cluster config
+		publishNodeDrainStatus(&zedkubeCtx, kubeapi.NOTREQUESTED)
+	}
+
```

Comment:
when the node is transition from single node into cluster mode later, it needs to handle the status correctly

Reply:
it's probably better if in non-cluster mode, then not to publish this item

Reply:
I'll remove that publish

Reply:
In non-cluster mode it publishes kubeapi.NOTSUPPORTED so that the callers to GetNodeDrainStatus() don't call RequestNodeDrain().

Reply:
but can the non-existence of 'drainstatus' be in the same treatment?
also from above original comment, this is done at initial stage of zedkube starts, if the cluster mode changes in the middle, this needs to be changed in the status.

Reply:
In zedkube.go handleEdgeNodeClusterConfigImpl() just after switching from single->cluster the NodeDrainStatus is set to NOTREQUESTED.

Reply:
@naiming-zededa do you mean remove the 'NOTSUPPORTED' enum status and instead unpublish or never publish NodeDrainStatus global id in that case?

Reply:
Right, that seems to be clean, since most devices don't run in cluster mode. Is that a big change?

Reply:
On non kubevirt builds it is already not published.  So this is just removing the publish for single-node kubevirt builds, I'll review it.

Reply:
Right, just for single node which is most kubevirt image usage I would think. So no publication is the same as unotsupport i guess